gnssanalysis/gn_download.py

```diff
@@ -268,8 +292,13 @@ def check_n_download_url(url, dwndir, filename=False):
         out_f = _Path(dwndir) / filename
         download_url(url, out_f)
 
+    if uncomp:
+        _subprocess.run(["uncompress", f"{out_f}"])
```

Comment:
Feeding effectively an arbitrary string to `subprocess.run` makes me a little uncomfortable from a security perspective. I'm certainly not an expert on how `subprocess.run` works and I'm pretty sure it's not parsed by a shell, which makes things a little better, but part of my mind can't help but wonder if specially crafted input could exploit this.

Reply:
I think we're safe from escaping, going by: https://docs.python.org/3/library/subprocess.html#security-considerations

We may also be additionally protected **_if_** `Path` validates the path when joining with the (overloaded?) `/` operator... presumably that would make it harder to sneak an escape in there.